src/hdmf/backends/hdf5/h5tools.py

```diff
@@ -54,10 +55,13 @@ def __init__(self, **kwargs):
         path, manager, mode, comm, file_obj, driver = popargs('path', 'manager', 'mode', 'comm', 'file', 'driver',
                                                               kwargs)
 
+        if path is None and file is None:
+            raise ValueError("You must supply either a path or a file.")
```

Comment:
Can you please add a unit test for this additional error case. Otherwise this PR looks good to me.